modules/nf-core/rtgtools/svdecompose/main.nf

```diff
@@ -8,28 +8,33 @@ process RTGTOOLS_SVDECOMPOSE {
         'biocontainers/rtg-tools:3.12.1--hdfd78af_0' }"
 
     input:
-    tuple val(meta), path(vcf), path(tbi)
+    tuple val(meta), path(input), path(tbi)
 
     output:
-    tuple val(meta), path("*.vcf.gz"), emit: bnd_vcf
-    path "versions.yml"              , emit: versions
+    tuple val(meta), path("*.vcf.gz")    , emit: vcf
+    tuple val(meta), path("*.vcf.gz.tbi"), emit: index
+    path "versions.yml"                  , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
-    def args = task.ext.args ?: ""
-    def prefix = task.ext.prefix ?: "${meta.id}"
-    def index = tbi ? "" : "rtg index ${vcf}"
+    def args      = task.ext.args ?: ""
+    def prefix    = task.ext.prefix ?: "${meta.id}"
+    def index_vcf = tbi ? "" : "rtg index ${input}"
     def avail_mem = task.memory.toGiga() + "G"
+    if ("$input" == "${prefix}.vcf.gz") error "Input and output names are the same, set prefix in module configuration to disambiguate!"
+
     """
-    ${index}
+    ${index_vcf}
 
     rtg RTG_MEM=$avail_mem svdecompose \\
         ${args} \\
-        --input=${vcf} \\
+        --input=${input} \\
         --output=${prefix}.vcf.gz
 
+    rtg index ${prefix}.vcf.gz
```

Comment:
@kubranarci the `rtg` commands automatically index all compressed output files -- doing it explicitly just wastes time.